package.json

```diff
@@ -32,34 +31,34 @@
   "peerDependencies": {
     "eslint": ">=8 <9"
   },
+  "sideEffects": false,
   "devDependencies": {
-    "@checkdigit/jest-config": "^3.1.1",
-    "@checkdigit/prettier-config": "^5.0.0",
-    "@checkdigit/typescript-config": "^6.0.0",
-    "@types/eslint": "^8.44.9",
-    "@typescript-eslint/eslint-plugin": "^6.15.0",
-    "@typescript-eslint/parser": "^6.15.0",
+    "@checkdigit/jest-config": "^6.0.0",
+    "@checkdigit/prettier-config": "^5.3.0",
+    "@checkdigit/typescript-config": "^7.0.0",
+    "@types/eslint": "^8.56.6",
+    "@typescript-eslint/eslint-plugin": "^7.3.1",
+    "@typescript-eslint/parser": "^7.3.1",
     "eslint-config-prettier": "^9.1.0",
-    "eslint-plugin-eslint-plugin": "^5.2.1",
+    "eslint-plugin-eslint-plugin": "^5.4.1",
     "eslint-plugin-import": "^2.29.1",
     "eslint-plugin-no-only-tests": "^3.1.0",
     "eslint-plugin-no-secrets": "^0.8.9",
     "eslint-plugin-node": "^11.1.0",
-    "eslint-plugin-sonarjs": "^0.23.0"
+    "eslint-plugin-sonarjs": "^0.24.0"
   },
   "scripts": {
-    "prepublishOnly": "npm run build:dist-types && npm run build:dist-cjs && npm run build:dist-mjs",
+    "prepublishOnly": "npm run build:dist-types && npm run build:dist-mjs",
     "build:dist-types": "rimraf dist-types && npx builder --type=types --outDir=dist-types",
-    "build:dist-cjs": "rimraf dist-cjs && npx builder --type=commonjs --sourceMap --entryPoint=index.ts --outDir=dist-cjs --outFile=index.cjs && echo \"module.exports = module.exports.default;\" >> dist-cjs/index.cjs && node dist-cjs/index.cjs",
```

Comment:
one thing I just noticed - you might want to check if eslint works with a pure ESM plugin.  maybe just try the beta out with eslint-config first before squerging this?

Reply:
Should we convert eslint-config to use esm to test eslint-plugin since there will be discrepancies loading modules